### DISCUSSION TITLE
FileUpload component - issue with delete uploaded file

### DISCUSSION THREAD

**Discussion by tbdev-icm** [filamentphp/filament#14316] [Help]
### Package

Form builder

### Package Version

v3.2.57

### How can we help you?

I have a simple file upload component in my [Model]Resource:

```
return $form->schema([
                TextInput::make('title'),
                TextInput::make('slug'),
                FileUpload::make('avatar')
                    ->disk('s3')
                    ->directory($avatarPath),
                RichEditor::make('content'),
            ]);
```

This generally works fine – the file uploads to S3 and is saved in the database. However, when I click the delete button, I encounter the following error:

`Filament\Forms\Components\BaseFileUpload::deleteUploadedFile(): Argument #1 ($fileKey) must be of type Closure, string given, called in /var/www/html/vendor/filament/forms/src/Concerns/SupportsFileUploadFields.php on line 15`

It looks like an issue with argument types in the methods within the vendor directory...

I'm using Laravel 11.

Can anyone help with this? 🥹

**Comment by leandrocfe** [comment]
I saw this topic on Discord. I’ve never encountered this issue before. Maybe try running it locally using MinIO to see if the issue also occurs.

**Reply by tbdev-icm** [reply]
Unfortunately, the issue here isn't the storage... (I tried ;))

**Reply by tbdev-icm** [reply]
The code that generates the component:
```
deleteUploadedFileUsing: async (fileKey) => {
                        return await $wire.deleteUploadedFile('data.avatar', fileKey)
                    },
```
And this is that method...
```
 public function deleteUploadedFile(\Closure $fileKey): static
 ```
